src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs

```diff
@@ -1609,9 +1605,10 @@ private void PopulateKnownTypes()
                 AddTypeIfNotNull(_knownTypes, _jsonValueType);
                 AddTypeIfNotNull(_knownTypes, _jsonDocumentType);
 
-                _knownUnsupportedTypes.Add(_serializationInfoType);
-                _knownUnsupportedTypes.Add(_intPtrType);
-                _knownUnsupportedTypes.Add(_uIntPtrType);
+                AddTypeIfNotNull(_knownUnsupportedTypes, _serializationInfoType);
+                AddTypeIfNotNull(_knownUnsupportedTypes, _serializationInfoType);
```

Comment:
_serializationInfoType seem to be added twice

Reply:
Good catch, thanks